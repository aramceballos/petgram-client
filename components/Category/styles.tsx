import styled from 'styled-components'

type TContainer = {
  onClick: () => void
}

export const Container = styled.div<TContainer>`
  border-radius: 50%;
  background: linear-gradient(45deg, orange, #962b96);
  padding: 2px;
  width: 72px;
  height: 72px;
`
export const Image = styled.img`
  background-color: #fff;
  border: 2px solid #fff;
  border-radius: 50%;
  height: auto;
  overflow: hidden;
  object-fit: cover;
  height: 68px;
  width: 68px;
`
