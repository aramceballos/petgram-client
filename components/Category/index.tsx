import React, { useState } from 'react'
import axios from 'axios'
import { useCookies } from 'react-cookie'

import { Container, Image } from './styles'
import Story from '../Story'

const DEFAULT_IMAGE = 'https://i.imgur.com/dJa0Hpl.jpg'

const Category = ({ image_url = DEFAULT_IMAGE, id }: ICategory) => {
  const [open, setOpen] = useState(false)
  const [loading, setLoading] = useState(false)
  const [currentStoryIndex, setCurrentStoryIndex] = useState(0)
  const [stories, setStories] = useState<IPost[]>([])
  const [cookies] = useCookies(['t'])
  const token = cookies['t']

  const handleClick = async () => {
    setOpen(true)

    try {
      setLoading(true)
      const res = await axios('http://localhost:5000/api/p', {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      })
      const filteredResults = res.data.data.filter(
        (item: IPost) => item.category_id === id
      )
      setStories(filteredResults)
      setLoading(false)
    } catch (error) {
      if (
        error.response?.data?.message === 'Missing or malformed JWT' ||
        error.response?.data?.message === 'Invalid or expired JWT'
      ) {
        cookies.set('t')
      }
      console.error(error.response?.data?.message)
    }
    setLoading(false)
  }

  const handleClose = () => {
    setOpen(false)
  }

  const handleNext = () => {
    if (currentStoryIndex === stories.length - 1) {
      handleClose()
    } else {
      setCurrentStoryIndex(currentStoryIndex + 1)
    }
  }

  const handleBack = () => {
    if (currentStoryIndex === 0) {
      handleClose()
    } else {
      setCurrentStoryIndex(currentStoryIndex - 1)
    }
  }

  return (
    <>
      <Container onClick={handleClick}>
        <Image src={image_url} alt="category image" />
      </Container>
      <Story
        open={open}
        loading={loading}
        story={stories[currentStoryIndex]}
        onClose={handleClose}
        onNext={handleNext}
        onBack={handleBack}
        stories={stories}
        currentStoryIndex={currentStoryIndex}
      />
    </>
  )
}

export default Category
