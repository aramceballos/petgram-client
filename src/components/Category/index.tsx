import React from 'react'
import { Link, Image } from './styles'

const DEFAULT_IMAGE = 'https://i.imgur.com/dJa0Hpl.jpg'

const Category = ({ image_url = DEFAULT_IMAGE, path = '', category }) => (
  <Link to={path}>
    <Image src={image_url} alt="category image" />
    {category}
  </Link>
)

export default Category
