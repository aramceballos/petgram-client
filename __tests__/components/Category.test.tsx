import React from 'react'
import { shallow } from 'enzyme'
import Category from '../../components/Category'

describe('<Category />', () => {
  test('should render correctly with no data', () => {
    const props = {
      id: 0,
      category: '',
      image_url: '',
    }
    const component = shallow(<Category {...props} />)
    expect(component).toMatchSnapshot()
  })

  test('should render correctly with data', () => {
    const props = {
      id: 5,
      category: 'test_category',
      image_url: 'https://i.imgur.com/dJa0Hpl.jpg',
    }
    const component = shallow(<Category {...props} />)
    expect(component).toMatchSnapshot()
  })
})
