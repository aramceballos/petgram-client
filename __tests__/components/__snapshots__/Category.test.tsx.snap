// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> should render correctly with data 1`] = `ShallowWrapper {}`;

exports[`<Category /> should render correctly with no data 1`] = `ShallowWrapper {}`;
